proposals/automating-releases.md

```diff
@@ -0,0 +1,185 @@
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created [for the website][operator-sdk-doc]
+
+## Open Questions
+
+1. We currently build images for an architecture on machines of that architecture.
+This is not a requirement for Go binaries themselves, but might be for Ansible/Helm
+dependencies since they install binaries for particular architectures. What are
+these dependencies and can they be installed in a container during image build?
+1. This proposal encompasses all binaries/images currently built for an operator-sdk release.
+Once Ansible- and Helm-related code are split into their respective repos, will
+this release process follow?
+1. Continue using our [changelog generator][sdk-changelog-generator], or use goreleaser's
+[generator][goreleaser-changelog]?
+
+## Summary
+
+This proposal outlines an automated release pipeline for the operator-sdk repository
+that includes both binary and image builds. The [`goreleaser`][goreleaser] release tool
+will replace the existing combination of the release script `release.sh`, manual `git`
+steps, and manual release publishing with a one "click" solution. To users, no changes
+will occur since the release artifacts are not changing. To developers, the release
+process will become much simpler, hopefully to the point where a new contributor can
+run the release with little process knowledge.
+
+## Motivation
+
+Releasing the operator-sdk repo currently requires several steps that are mostly manual,
+with some automation around building binaries for all architectures supported. Tools like
+[`goreleaser`][goreleaser] can automate this process entirely by tagging code, and building
+and publishing binaries and images with one "click".
+
+## Goals
+
+- Remove all manual steps from the release process
+- Centralize release configuration
+- One-click releasing
+
+### Non-Goals
+
+- Change the content or constituency of output release artifacts in any way
+- Change the testing process in any way
+- Change our CI provider
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - Release the operator-sdk repo in one click
+
+As a contributor I want to run the release in as few operations as possible.
+This feature will enable new contributors to release the repo easily, instead
+of relying on a long onboarding document and the expertise of a few contributors.
+
+#### Story 2 - Easily modify release configuration
+
+As a contributor I want to easily improve the release configuration whenever necessary,
+which includes adding new release architectures. Configuration should be documented,
+comprehensible, and centralized.
+
+### Implementation Details/Notes/Constraints
+
+The [`goreleaser`][goreleaser] release tool satisfies the above goals and user stories:
+- Manages code tags
```

Comment:
How does it do this? Do we run goreleaser with a tag and it works?

Reply:
Goreleaser requires tags on the commit it is running against unless the `--snapshot` flag is included in the invocation.

Worthwhile reading: https://goreleaser.com/quick-start/

Reply:
Updated